tests/tests/util/findAvailablePorts.ts

```diff
@@ -0,0 +1,32 @@
+import tcpPortUsed from "tcp-port-used";
+
+export async function findAvailablePorts() {
```

Comment:
I would design it differently. The risk here comes from the fact it is checking the ports incrementally, starting from the same number each time, the race condition are more likely to happen in that case. It is also not efficient.

Using a random port might be more effective

Reply:
(this is for the parallel mode, not the serial one)

Reply:
You're right. Turns out this isn't exactly a trivial problem... I'll leave this link here: https://github.com/mochajs/mocha/issues/4456